tools/azure-sdk-tools/devtools_testutils/proxy_docker_startup.py

```diff
@@ -0,0 +1,136 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import json
+import os
+import logging
+import shlex
+import sys
+import time
+from typing import TYPE_CHECKING
+
+import pytest
+import subprocess
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+
+_LOGGER = logging.getLogger()
+
+CONTAINER_NAME = "ambitious_azsdk_test_proxy"
+LINUX_IMAGE_SOURCE_PREFIX = "azsdkengsys.azurecr.io/engsys/testproxy-lin"
+WINDOWS_IMAGE_SOURCE_PREFIX = "azsdkengsys.azurecr.io/engsys/testproxy-win"
+
+REPO_ROOT = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "..", "..", ".."))
+
+
+def get_image_tag():
+    # type: () -> str
+    """Gets the test proxy Docker image tag from the docker-start-proxy.ps1 script in /eng/common"""
+    pwsh_script_location = os.path.abspath(
+        os.path.join(REPO_ROOT, os.path.relpath("eng/common/testproxy/docker-start-proxy.ps1"))
+    )
+
+    image_tag = None
+    with open(pwsh_script_location, "r") as f:
+        for line in f:
+            if line.startswith("$SELECTED_IMAGE_TAG"):
+                image_tag_with_quotes = line.split()[-1]
+                image_tag = image_tag_with_quotes.strip('"')
+
+    return image_tag
+
+
+def get_container_info():
+    # type: () -> Optional[dict]
+    """Returns a dictionary containing the test proxy container's information, or None if the container isn't present"""
+    proc = subprocess.Popen(
+        shlex.split("docker container ls -a --format '{{json .}}' --filter name=" + CONTAINER_NAME),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    output, stderr = proc.communicate()
+    try:
+        return json.loads(output)
+    # We'll get a JSONDecodeError on Py3 (ValueError on Py2) if output is empty (i.e. there's no proxy container)
+    except ValueError:
+        return None
```

Comment:
If could log/print out the error details would be helpful while debugging I think.

Reply:
The details of this are logged in `start_test_proxy` since that's where we interpret the output of `get_container_info`, but I added some comments to make the behavior more clear